ksqldb-engine/src/main/java/io/confluent/ksql/engine/KsqlEngine.java

```diff
@@ -186,6 +199,11 @@ public KsqlEngine(
     this.metricCollectors = metricCollectors;
 
     cleanupService.startAsync();
+    if (getTransientQueryCleanupServiceEnabled(ksqlConfig)) {
+      this.transientQueryCleanupService
```

Comment:
It would be nice if you could set this in the constructor, but I see that there's a bit of a circular dependency.  Maybe just put a comment in the setter about how it's being used.

Reply:
will take care of it in the follow-up